install/installer/pkg/config/v1/config.go

```diff
@@ -172,6 +174,9 @@ type Config struct {
 
 	Certificate ObjectRef `json:"certificate" validate:"required"`
 
+	// Name of the kubernetes object to use for signature of Personal Access Tokens
+	PersonalAccessTokenSigningKey ObjectRef `json:"personalAccessTokenSigningKey" validate:"required"`
```

Comment:
How do we communicate what properties must exist on the `SecretRef`? Here we say that this secret must exist, but we also require that the secret also contains a particular key, which has a value of the PAT Signing Key

Reply:
>Here we say that this secret must exist

I don't think we communicate this. We expect the field `personalAccessTokenSigningKey` to be set to some value (there is no run-time verification whether it's there) and considering we have a [default value for this](https://github.com/gitpod-io/gitpod/pull/15059/files#diff-ed0dab0d7e4e73c6487f629ed17403feeb8ee3b8cf1601e5c940f7a9365ccf17R58), This will never fail. The deployment will eventually fail if the secret does not exist as it's no longer optional.

The reason for having a default value is not to change the existing behavior as the new installer would fail wherever this field is not set 🤔  

> we also require that the secret also contains a particular key, which has a value of the PAT Signing Key

This was easy for certificates as there is a standard format for storing certs in Kubernetes secret. I don't think we have a straightforward way of doing this. We can add a comment and [render it in the docs](https://github.com/gitpod-io/gitpod/blob/main/install/installer/docs/overview.md#config) on how we expect this to be present.

cc: @mrsimonemms WDYT? 🤔